pkg/pod/limitrange/transformer.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package limitrange
+
+import (
+	"context"
+
+	"github.com/tektoncd/pipeline/pkg/pod"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/client-go/kubernetes"
+)
+
+var resourceNames = []corev1.ResourceName{corev1.ResourceCPU, corev1.ResourceMemory, corev1.ResourceEphemeralStorage}
+
+func isZero(q resource.Quantity) bool {
+	return (&q).IsZero()
+}
+
+func NewTransformer(ctx context.Context, namespace string, clientset kubernetes.Interface) pod.Transformer {
+	return func(p *corev1.Pod) (*corev1.Pod, error) {
+		limitRange, err := getVirtualLimitRange(ctx, namespace, clientset)
+		if err != nil {
+			return p, err
+		}
+		// No LimitRange defined, nothing to transform, bail early we don't have anything to transform.
+		if limitRange == nil {
+			return p, nil
+		}
+
+		// The assumption here is that the min, max, default, ratio have already been
+		// computed if there is multiple LimitRange to satisfy the most (if we can).
+		// Count the number of containers (that we know) in the Pod.
+		// This should help us find the smallest request to apply to containers
+		// We are adding +1 to the number of container to take into account the init containers.
+		// The reason to use +1 only is that, k8s tread request on all init container as one (getting the max of all)
+		nbContainers := len(p.Spec.Containers) + 1
+		// FIXME(vdemeester) maxLimitRequestRatio to support later
```

Comment:
could we link to an issue vs just your username? (otherwise this kinda thing is hard to follow up on later on)

Reply:
Yes, kept this here but indeed I intended to create an issue about this 👼🏼

Reply:
https://github.com/tektoncd/pipeline/issues/4230 👼🏼